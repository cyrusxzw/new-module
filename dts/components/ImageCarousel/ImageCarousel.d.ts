export function ImageCarousel({
  autoplaySpeed,
  className,
  hasAutoplay,
  hasFlushPagination,
  isCompact,
  slides,
  theme,
}: {
  autoplaySpeed: any;
  className: any;
  hasAutoplay: any;
  hasFlushPagination: any;
  isCompact: any;
  slides: any;
  theme: any;
}): JSX.Element;
export namespace ImageCarousel {
  namespace propTypes {
    const autoplaySpeed: PropTypes.Requireable<number>;
    const className: PropTypes.Requireable<string>;
    const hasAutoplay: PropTypes.Requireable<boolean>;
    const hasFlushPagination: PropTypes.Requireable<boolean>;
    const isCompact: PropTypes.Requireable<boolean>;
    const slides: PropTypes.Requireable<
      PropTypes.InferProps<{
        caption: PropTypes.Requireable<string>;
        id: PropTypes.Requireable<string>;
        image: PropTypes.Validator<object>;
      }>[]
    >;
    const theme: PropTypes.Requireable<string>;
  }
  namespace defaultProps {
    const autoplaySpeed_1: number;
    export { autoplaySpeed_1 as autoplaySpeed };
    const className_1: any;
    export { className_1 as className };
    const hasAutoplay_1: boolean;
    export { hasAutoplay_1 as hasAutoplay };
    const hasFlushPagination_1: boolean;
    export { hasFlushPagination_1 as hasFlushPagination };
    const isCompact_1: boolean;
    export { isCompact_1 as isCompact };
    const slides_1: any[];
    export { slides_1 as slides };
    const theme_1: string;
    export { theme_1 as theme };
  }
}
import PropTypes from 'prop-types';
