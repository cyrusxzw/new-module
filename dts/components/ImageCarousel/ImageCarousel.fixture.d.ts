export namespace ImageCarouselFixture {
  const autoplaySpeed: number;
  const hasAutoplay: boolean;
  const slides: {
    caption: string;
    image: {
      altText: string;
      small: string;
      medium: string;
      large: string;
    };
  }[];
  const theme: string;
}
