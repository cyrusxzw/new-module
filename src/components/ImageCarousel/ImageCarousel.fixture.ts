const ImageCarouselFixture = {
  autoplaySpeed: 3000,
  hasAutoplay: false,
  slides: [
    {
      caption:
        'Nunc dignissim quis arcu a lobortis. Nullam auctor id nulla non sagittis. Quisque non enim efficitur quam mattis semper a nec arcu.',
      image: {
        altText: 'Nunc dignissim',
        small: './assets/images/ImageCarousel/image-two-small.jpg',
        medium: './assets/images/ImageCarousel/image-two-medium.jpg',
        large: './assets/images/ImageCarousel/image-two-large.jpg',
      },
    },
    {
      caption:
        'Vivamus sollicitudin ac ligula nec convallis. Etiam varius faucibus libero id congue.',
      image: {
        altText: 'Vivamus sollicitudin',
        small: './assets/images/ImageCarousel/image-three-small.jpg',
        medium: './assets/images/ImageCarousel/image-three-medium.jpg',
        large: './assets/images/ImageCarousel/image-three-large.jpg',
      },
    },
  ],
  theme: 'dark',
};

export { ImageCarouselFixture };
