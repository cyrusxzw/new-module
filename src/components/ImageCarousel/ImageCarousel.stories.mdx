import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import { StorybookGridOverlay } from '~.storybook/components';
import { TwoColumnLayout } from '~/components/TwoColumnLayout';
import { BodyCopy } from '~/components/BodyCopy';
import { Podium } from '~/components/Podium';
import { ImageCarousel } from './ImageCarousel';
import { ImageCarouselFixture } from './ImageCarousel.fixture';

<Meta title="Components/ImageCarousel" component={ImageCarousel} />

# ImageCarousel

<Canvas>
  <Story name="Base component">
    {(() => {
      const autoplaySpeed = knobs.number(
        'autoplaySpeed',
        ImageCarouselFixture.autoplaySpeed,
        'Presentation',
      );
      const hasAutoplay = knobs.boolean('hasAutoplay', false, 'Presentation');
      const hasFlushPagination = knobs.boolean(
        'hasFlushPagination',
        false,
        'Presentation',
      );
      const isCompact = knobs.boolean('isCompact', false, 'Presentation');
      const slides = [
        {
          caption: knobs.text(
            'caption',
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum facilisis ligula elementum.',
            'First slide copy',
          ),
          image: {
            altText: 'Lorem ipsum dolor sit amet',
            sizes: {
              small: './assets/images/ImageCarousel/image-one-small.jpg',
              medium: './assets/images/ImageCarousel/image-one-medium.jpg',
              large: './assets/images/ImageCarousel/image-one-large.jpg',
            },
          },
        },
        ...ImageCarouselFixture.slides,
      ];
      const theme = knobs.select(
        'theme',
        ['dark', 'light'],
        'dark',
        'Presentation',
      );
      const implementation = {
        gridOverlay: {
          isActive: knobs.boolean('Show grid overlay', false, 'Implementation'),
          hasInvertedColours: knobs.boolean(
            'Invert grid overlay colours',
            false,
            'Implementation',
          ),
        },
      };
      return (
        <>
          <ImageCarousel
            autoplaySpeed={autoplaySpeed}
            hasAutoplay={hasAutoplay}
            hasFlushPagination={hasFlushPagination}
            isCompact={isCompact}
            slides={slides}
            theme={theme}
          />
          <StorybookGridOverlay
            isActive={implementation.gridOverlay.isActive}
            hasInvertedColours={implementation.gridOverlay.hasInvertedColours}
          />
        </>
      );
    })()}
  </Story>
</Canvas>

## API

<ArgsTable of={ImageCarousel} />

## Variations

### With Autoplay

<Canvas>
  <Story name="With Autoplay">
    <ImageCarousel
      autoplaySpeed={knobs.number(
        'autoplaySpeed',
        ImageCarouselFixture.autoplaySpeed,
        'Presentation',
      )}
      hasAutoplay={true}
      slides={ImageCarouselFixture.slides}
    />
  </Story>
</Canvas>

### Rendered in Two Column Layout

Wrapped in a `Podium` component. The `ImageCarousel` component has `hasAutoplay` enabled.

<Canvas>
  <Story name="Rendered in Two Column Layout">
    <Podium
      verticalPadding="small"
      paddingRight="small"
      isHorizontalFlushOnSmall={true}
      isHorizontalFlushOnMedium={true}
    >
      <TwoColumnLayout
        hasFullWidthContent={knobs.boolean(
          'hasFullWidthContent',
          true,
          'Presentation',
        )}
        id="TwoColumn-id"
        isFlushOnSmall={knobs.boolean('isFlushOnSmall', true, 'Presentation')}
        isReversed={knobs.boolean('isReversed', true, 'Presentation')}
        sidebar={
          <BodyCopy
            cta={{
              text: 'Quisque non enim',
              url: '/',
              style: 'Internal Button Link',
            }}
            ctaTextAlign="left"
            heading="Nunc dignissim quis"
            copy="Nunc dignissim quis arcu a lobortis. Nullam auctor id nulla non sagittis. Quisque non enim efficitur quam mattis."
          />
        }
        content={
          <ImageCarousel
            hasAutoplay={true}
            hasFlushPagination={true}
            isCompact={true}
            slides={ImageCarouselFixture.slides}
          />
        }
      />
    </Podium>
  </Story>
</Canvas>
