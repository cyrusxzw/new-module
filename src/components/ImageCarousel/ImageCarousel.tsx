import React from 'react';
import { Carousel } from '~/components/Carousel';
import type { ImageCarouselType } from './ImageCarousel.types';

const ImageCarousel: ImageCarouselType = ({
  autoplaySpeed = 3000,
  className,
  hasAutoplay = false,
  hasFlushPagination = false,
  isCompact = false,
  slides = [],
  theme,
}) => (
  <Carousel
    autoplaySpeed={autoplaySpeed}
    className={className}
    hasAutoplay={hasAutoplay}
    hasFlushPagination={hasFlushPagination}
    hasFullWidthSlides={true}
    hasShowCaption={true}
    hasSlideCounter={true}
    isCompact={isCompact}
    slides={slides}
    theme={theme}
  />
);

export { ImageCarousel };
