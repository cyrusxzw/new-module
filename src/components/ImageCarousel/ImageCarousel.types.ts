import type { Themes } from '~/types';
import type { ImageProps } from '../Image/Image.types';

type Slide = {
  caption?: string;
  id?: string;
  image: ImageProps;
};

type ImageCarouselProps = {
  autoplaySpeed?: number;
  className?: string;
  hasAutoplay?: boolean;
  hasFlushPagination?: boolean;
  isCompact?: boolean;
  slides?: Slide[];
  theme?: Themes;
};

export type { ImageCarouselProps };
