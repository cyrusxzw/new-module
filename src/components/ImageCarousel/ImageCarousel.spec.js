import React from 'react';
import { render } from '@testing-library/react';
import { ImageCarousel } from './ImageCarousel';
import { ImageCarouselFixture } from './ImageCarousel.fixture';

describe('<ImageCarousel />', () => {
  it('should render base component correctly', () => {
    const { container } = render(
      <ImageCarousel slides={ImageCarouselFixture.slides} />,
    );

    expect(container).toMatchSnapshot();
  });
});
