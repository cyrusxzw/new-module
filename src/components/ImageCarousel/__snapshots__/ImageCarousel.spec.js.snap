// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageCarousel /> should render base component correctly 1`] = `
<div>
  <div
    class="base fullWidthSlides dark"
  >
    <div
      class="slick-slider carousel slick-initialized"
      dir="ltr"
    >
      <button
        class="base dark base"
        title="next slide"
        type="button"
      >
        <svg
          aria-labelledby="chevron-1"
          class="base chevron dark"
          data-testid="data-testid-Icon"
          focusable="false"
          height="16"
          role="img"
          style="height: 16px; width: 16px;"
          tabindex="-1"
          viewBox="0 0 50 50"
          width="16"
        >
          <title
            id="chevron-1"
          >
            chevron
          </title>
          <g>
            <polygon
              class="chevron"
              points="25,31.3 4.2,10.5 0.1,14.6 25,39.5 50,14.6 45.9,10.5"
            />
          </g>
        </svg>
      </button>
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="-1"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="slideWrapper"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <div
                  class="item"
                >
                  <div
                    class="imageContainer"
                  >
                    <picture
                      class="base picture dark image fullWidthSlide"
                    >
                      <source
                        media="(min-width: 1025px)"
                        srcset="./assets/images/ImageCarousel/image-three-large.jpg"
                      />
                      <source
                        media="(min-width: 640px)"
                        srcset="./assets/images/ImageCarousel/image-three-medium.jpg"
                      />
                      <source
                        media="(min-width: 0px)"
                        srcset="./assets/images/ImageCarousel/image-three-small.jpg"
                      />
                      <img
                        alt="Vivamus sollicitudin"
                        loading="lazy"
                      />
                    </picture>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="slideWrapper"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <div
                  class="item"
                >
                  <div
                    class="imageContainer"
                  >
                    <picture
                      class="base picture dark image fullWidthSlide"
                    >
                      <source
                        media="(min-width: 1025px)"
                        srcset="./assets/images/ImageCarousel/image-two-large.jpg"
                      />
                      <source
                        media="(min-width: 640px)"
                        srcset="./assets/images/ImageCarousel/image-two-medium.jpg"
                      />
                      <source
                        media="(min-width: 0px)"
                        srcset="./assets/images/ImageCarousel/image-two-small.jpg"
                      />
                      <img
                        alt="Nunc dignissim"
                        loading="lazy"
                      />
                    </picture>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="1"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="slideWrapper"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <div
                  class="item"
                >
                  <div
                    class="imageContainer"
                  >
                    <picture
                      class="base picture dark image fullWidthSlide"
                    >
                      <source
                        media="(min-width: 1025px)"
                        srcset="./assets/images/ImageCarousel/image-three-large.jpg"
                      />
                      <source
                        media="(min-width: 640px)"
                        srcset="./assets/images/ImageCarousel/image-three-medium.jpg"
                      />
                      <source
                        media="(min-width: 0px)"
                        srcset="./assets/images/ImageCarousel/image-three-small.jpg"
                      />
                      <img
                        alt="Vivamus sollicitudin"
                        loading="lazy"
                      />
                    </picture>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="2"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="slideWrapper"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <div
                  class="item"
                >
                  <div
                    class="imageContainer"
                  >
                    <picture
                      class="base picture dark image fullWidthSlide"
                    >
                      <source
                        media="(min-width: 1025px)"
                        srcset="./assets/images/ImageCarousel/image-two-large.jpg"
                      />
                      <source
                        media="(min-width: 640px)"
                        srcset="./assets/images/ImageCarousel/image-two-medium.jpg"
                      />
                      <source
                        media="(min-width: 0px)"
                        srcset="./assets/images/ImageCarousel/image-two-small.jpg"
                      />
                      <img
                        alt="Nunc dignissim"
                        loading="lazy"
                      />
                    </picture>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="3"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="slideWrapper"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <div
                  class="item"
                >
                  <div
                    class="imageContainer"
                  >
                    <picture
                      class="base picture dark image fullWidthSlide"
                    >
                      <source
                        media="(min-width: 1025px)"
                        srcset="./assets/images/ImageCarousel/image-three-large.jpg"
                      />
                      <source
                        media="(min-width: 640px)"
                        srcset="./assets/images/ImageCarousel/image-three-medium.jpg"
                      />
                      <source
                        media="(min-width: 0px)"
                        srcset="./assets/images/ImageCarousel/image-three-small.jpg"
                      />
                      <img
                        alt="Vivamus sollicitudin"
                        loading="lazy"
                      />
                    </picture>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="base dark base"
        title="next slide"
        type="button"
      >
        <svg
          aria-labelledby="chevron-2"
          class="base chevron dark"
          data-testid="data-testid-Icon"
          focusable="false"
          height="16"
          role="img"
          style="height: 16px; width: 16px;"
          tabindex="-1"
          viewBox="0 0 50 50"
          width="16"
        >
          <title
            id="chevron-2"
          >
            chevron
          </title>
          <g>
            <polygon
              class="chevron"
              points="25,31.3 4.2,10.5 0.1,14.6 25,39.5 50,14.6 45.9,10.5"
            />
          </g>
        </svg>
      </button>
      <div
        class="base"
        data-testid="data-testid-Carousel-Pagination"
      >
        <div
          class="list dark"
        >
          <div
            class="bar"
            style="width: 50%; left: 0%;"
          />
        </div>
      </div>
    </div>
    <footer
      class="footer"
    >
      <div
        class="slideCounter"
      >
        1 / 2
      </div>
      <div
        class="caption"
      >
        Nunc dignissim quis arcu a lobortis. Nullam auctor id nulla non sagittis. Quisque non enim efficitur quam mattis semper a nec arcu.
      </div>
    </footer>
  </div>
</div>
`;
